cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -247,26 +253,34 @@ public static void sanityCheckCapacityEstimation(boolean allowCapacityEstimation
 
   /**
    * Rebalance the cluster
-   * @param goals the goals to be met during the rebalance. When empty all goals will be used.
-   * @param dryRun whether it is a dry run or not.
+   * @param goals The goals to be met during the rebalance. When empty all goals will be used.
+   * @param dryRun Whether it is a dry run or not.
    * @param requirements The cluster model completeness requirements.
-   * @param operationProgress the progress of the job to report.
+   * @param operationProgress The progress of the job to report.
    * @param allowCapacityEstimation Allow capacity estimation in cluster model if the requested broker capacity is unavailable.
    * @param concurrentPartitionMovements The maximum number of concurrent partition movements per broker
    *                                     (if null, use num.concurrent.partition.movements.per.broker).
    * @param concurrentLeaderMovements The maximum number of concurrent leader movements
    *                                  (if null, use num.concurrent.leader.movements).
-   * @return the optimization result.
-   * @throws KafkaCruiseControlException when the rebalance encounter errors.
+   * @param skipHardGoalCheck True if the provided {@code goals} do not have to contain all hard goals, false otherwise.
+   * @return The optimization result.
+   * @throws KafkaCruiseControlException When the rebalance encounter errors.
    */
   public GoalOptimizer.OptimizerResult rebalance(List<String> goals,
                                                  boolean dryRun,
                                                  ModelCompletenessRequirements requirements,
                                                  OperationProgress operationProgress,
                                                  boolean allowCapacityEstimation,
                                                  Integer concurrentPartitionMovements,
-                                                 Integer concurrentLeaderMovements) throws KafkaCruiseControlException {
-    GoalOptimizer.OptimizerResult result = getOptimizationProposals(goals, requirements, operationProgress, allowCapacityEstimation);
+                                                 Integer concurrentLeaderMovements,
+                                                 boolean skipHardGoalCheck) throws KafkaCruiseControlException {
+    GoalOptimizer.OptimizerResult result;
+    //If it is a PLE request, skip hard goal check.
+    if (goals != null && goals.size() == 1 && goals.get(0).equals(PreferredLeaderElectionGoal.class.getName())) {
```

Comment:
Shouldn't it be `PreferredLeaderElectionGoal.class.getName()` -> `PreferredLeaderElectionGoal.class.getSimpleName()`?